src/main/java/net/minecraftforge/registries/DataPackRegistriesHooks.java

```diff
@@ -5,34 +5,53 @@
 
 package net.minecraftforge.registries;
 
+import com.google.common.collect.ImmutableMap;
 import net.minecraft.core.Registry;
+import net.minecraft.core.RegistrySynchronization;
 import net.minecraft.resources.RegistryDataLoader;
 import net.minecraft.resources.ResourceKey;
 import org.jetbrains.annotations.ApiStatus;
 
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.HashSet;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 
 @ApiStatus.Internal
 public final class DataPackRegistriesHooks
 {
     private DataPackRegistriesHooks() {} // utility class
 
+    private static Map<ResourceKey<? extends Registry<?>>, RegistrySynchronization.NetworkedRegistryData<?>> NETWORKABLE_REGISTRIES = new LinkedHashMap<>();
     private static final List<RegistryDataLoader.RegistryData<?>> DATA_PACK_REGISTRIES = new ArrayList<>(RegistryDataLoader.WORLDGEN_REGISTRIES);
     private static final List<RegistryDataLoader.RegistryData<?>> DATA_PACK_REGISTRIES_VIEW = Collections.unmodifiableList(DATA_PACK_REGISTRIES);
     private static final Set<ResourceKey<? extends Registry<?>>> SYNCED_CUSTOM_REGISTRIES = new HashSet<>();
     private static final Set<ResourceKey<? extends Registry<?>>> SYNCED_CUSTOM_REGISTRIES_VIEW = Collections.unmodifiableSet(SYNCED_CUSTOM_REGISTRIES);
 
+    /* Internal forge hook for retaining mutable access to RegistryAccess's codec registry when it bootstraps. */
+    public static Map<ResourceKey<? extends Registry<?>>, RegistrySynchronization.NetworkedRegistryData<?>> grabNetworkableRegistries(ImmutableMap.Builder<ResourceKey<? extends Registry<?>>, RegistrySynchronization.NetworkedRegistryData<?>> builder)
+    {
+        if (!StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE).getCallerClass().equals(RegistrySynchronization.class))
+            throw new IllegalCallerException("Attempted to call DataPackRegistriesHooks#grabNetworkableRegistries!");
+        NETWORKABLE_REGISTRIES.forEach(builder::put);
+        NETWORKABLE_REGISTRIES = new HashMap<>(builder.build());
```

Comment:
Why is `NETWORKABLE_REGISTRIES` initialized to a `LinkedHashMap` and later reassigned to a `HashMap`? Why not both `LinkedHashMap`? Why not make the field `final`, and do `NETWORKABLE_REGISTRIES.clear()` then `NETWORKABLE_REGISTRIES.addAll(builder.build())` instead of reassigning the field?

Reply:
It was initially a `HashMap` but I changed it to a linked map to preserve insertion order. Although you do make a fair point that the map can be final.